src/main/java/dk/kb/image/TemplateMostUsedColors.java

```diff
@@ -45,9 +46,10 @@ List<DominantColorDto> getMostUsedColors(BufferedImage img, int x){
      * @param buckets list of bucket colors, type is defined by subclass.
      * @return the integer array bucketCounter, which contains the count for each bucket for the input image,
      */
-    int[] getBucketCount(BufferedImage img, List<C> buckets){
+    int[] getBucketCount(BufferedImage img, List<C> buckets) throws IOException {
         // Create bucket counter array
         int[] bucketCounter = new int[buckets.size()];
+        byte[] entriesForAllRgbColors = Thread.currentThread().getContextClassLoader().getResource("OklabBucketEntriesForAllRgbColors").openStream().readAllBytes();
```

Comment:
1. This should only be loaded once and kept in memory for subsequent calls
2. This is OK-Lab specific and should be in the `MostUsedOkLabColor` class

Reply:
Nearly there: You load `entriesForAllRgbColors` at the top of the `MostUsedOkLabColor`, but you do it once for every instance. If you make it static, it will be shared between instances.

Also: You forgot to remove loading of the same data in `calculateDistance`.